service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NonConsultationResourceMapper.java

```diff
@@ -0,0 +1,181 @@
+        return XpathExtractor.extractValuesForBloodPressure(component)
+            .altCode(BLOOD_PRESSURE_CODE_2);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildReferralRequest(String component) {
+        return XpathExtractor.extractValuesForReferralRequest(component)
+            .altCode(DEFAULT_CODE);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildDiagnosticRequest(String component) {
+        return null;
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildMedicationRequest(String component) {
+        return XpathExtractor.extractValuesForMedicationRequest(component)
+            .altCode(MEDICATION_REQUEST_CODE);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildCondition(String component) {
+        return XpathExtractor.extractValuesForCondition(component)
+            .altCode(CONDITION_CODE);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildProcedureRequest(String component) {
+        return XpathExtractor.extractValuesForProcedureRequest(component)
+            .altCode(DEFAULT_CODE);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildDocumentReference(String component) {
+        return XpathExtractor.extractValuesForDocumentReference(component)
+            .altCode(DEFAULT_CODE);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildQuestionnaireResponse(String component) {
+        return null;
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder buildObservation(String component, Observation observation) {
+        return mapObservation(observation, component);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder mapObservation(Resource resource, String component) {
+        Observation observation = (Observation) resource;
+        if (CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(EncounterComponentsMapper.NARRATIVE_STATEMENT_CODE))) {
+            return buildCommentObservation(component);
+        }
+        if (CodeableConceptMappingUtils.hasCode(observation.getCode(), EncounterComponentsMapper.BLOOD_CODES)) {
+            return buildBloodPressureObservation(component);
+        }
+
+        return buildUncategorisedObservation(component);
+    }
+
+    private EncounterTemplateParameters.EncounterTemplateParametersBuilder notMapped(String component) {
+        return null;
+    }
+
+    private boolean isMappableNonConsultationResource(Resource resource) {
+        if (resource.getResourceType().equals(ResourceType.Observation)) {
+            return true;
+        }
+        return resourceBuilder.containsKey(resource.getResourceType());
```

Comment:
Not sure if it's clearer, but for your consideration:
```suggestion
        return resource.getResourceType().equals(ResourceType.Observation)
            || resourceBuilder.containsKey(resource.getResourceType());
```